src/fiber_prompt/asyncify/asyncify_impl.c

```diff
@@ -140,14 +148,15 @@ __attribute__((noinline)) void fiber_free(fiber_t fiber) {
 // Yields control from within a fiber computation to whichever point
 // originally resumed the fiber.
 __attribute__((noinline)) void *fiber_yield_to(prompt_t prompt, void *arg) {
-  (void)prompt;
   if (active_fiber->state == YIELDING) {
     asyncify_stop_rewind();
     active_fiber->state = ACTIVE;
+    if (prompt != active_fiber->prompt) active_fiber->state = NAME_MISMATCH;
```

Comment:
why are you checking the prompt name here?

Reply:
I'd expect that you check the prompt name after `stop_unwind`. And if it doesn't match, then initiate another unwind.